platform/target_data.go

```diff
@@ -64,25 +40,14 @@ func GetTargetMetadata(fromImage imgutil.Image) (*files.TargetMetadata, error) {
 	return &tm, nil
 }
 
-// GetTargetOSFromFileSystem populates the target metadata you pass in if the information is available
-// returns a boolean indicating whether it populated any data.
-func GetTargetOSFromFileSystem(d fsutil.Detector, tm *files.TargetMetadata, logger log.Logger) {
-	if d.HasSystemdFile() {
-		contents, err := d.ReadSystemdFile()
-		if err != nil {
-			logger.Warnf("Encountered error trying to read /etc/os-release file: %s", err.Error())
-			return
-		}
-		info := d.GetInfo(contents)
-		if info.Version != "" || info.Name != "" {
-			tm.OS = "linux"
-			tm.Distro = &files.OSDistro{Name: info.Name, Version: info.Version}
-		}
-	}
-}
-
 // TargetSatisfiedForBuild treats empty fields as wildcards and returns true if all populated fields match.
-func TargetSatisfiedForBuild(base files.TargetMetadata, module buildpack.TargetMetadata) bool {
+func TargetSatisfiedForBuild(d fsutil.Detector, base files.TargetMetadata, module buildpack.TargetMetadata, logger log.Logger) bool {
+	// ensure we have all available data
+	if base.OS == "" || base.Distro == nil {
+		logger.Info("target distro name/version labels not found, reading /etc/os-release file")
```

Comment:
In what cases could `OS` be the empty string? If the author of the run image created it in such a way that the image didn't have properly specified metadata? The OCI spec suggests `OS` is REQUIRED:
https://github.com/opencontainers/image-spec/blob/main/config.md#properties

It's just at the moment, the log message (whose wording exists prior to this PR) isn't consistent with the conditional. ie: The message says "target distro name/version labels", but the conditional suggests it should say "OS metadata, or target distro name/version labels" etc. (But if `OS` should always be present, perhaps the `OS` check can be moved to an earlier assertion that errors instead?)

Reply:
Good point. I added an error in 609e287c